recipes/material-ui/index.ts

```diff
@@ -0,0 +1,443 @@
+    explanation: `@material-ui/core needs to be installed`,
+    packages: [{name: "@material-ui/core", version: "latest"}],
+  })
+  .addTransformFilesStep({
+    stepId: "modifyGetInitialPropsInCustomDocumentApp",
+    stepName: "Add custom getInitialProps logic in Custom Document",
+    explanation: `We will add custom getInitialProps logic in _document. We need to do this so that styles are correctly rendered on the server side.`,
+    singleFileSearch: paths.document(),
+    transform(ast: ASTNode, b: builders, t: NamedTypes) {
+      if (t.File.check(ast)) {
+        // import ServerStyleSheets
+        const serverStyleSheetsImport = b.importDeclaration(
+          [b.importSpecifier(b.identifier("ServerStyleSheets"))],
+          b.literal("@material-ui/core/styles"),
+        )
+
+        let isReactImported = false
+
+        visit(ast, {
+          visitImportDeclaration(path) {
+            // check if React is already imported
+            // if yes then we can skip importing it
+            // since we need it for useEffect
+            const source = path.value.source.value
+            if (source === "react") {
+              isReactImported = true
+
+              // currently, we only check if the default export is there
+              // because we use the hook as React.useEffect
+              // if not then add the default export
+              if (
+                !path.value.specifiers.some(
+                  (node: namedTypes.ImportDefaultSpecifier | namedTypes.ImportSpecifier) =>
+                    node.type === "ImportDefaultSpecifier",
+                )
+              ) {
+                path.value.specifiers.splice(0, 0, b.importDefaultSpecifier(b.identifier("React")))
+
+                return false
+              }
+            }
+            // check if DocumentContext is imported from blitz if not add that as well
+            else if (source === "blitz") {
+              if (
+                !path.value.specifiers
+                  .filter(
+                    (node: namedTypes.ImportDefaultSpecifier | namedTypes.ImportSpecifier) =>
+                      node.type === "ImportSpecifier",
+                  )
+                  .some(
+                    (node: namedTypes.ImportSpecifier) =>
+                      node?.imported?.name === "DocumentContext",
+                  )
+              ) {
+                path.value.specifiers.splice(
+                  0,
+                  0,
+                  b.importSpecifier(b.identifier("DocumentContext")),
+                )
+              }
+            }
+
+            return this.traverse(path)
+          },
+          visitClassBody(path) {
+            const {node} = path
+
+            const ctxParam = b.identifier("ctx")
+            ctxParam.typeAnnotation = b.tsTypeAnnotation(
+              b.tsTypeReference(b.identifier("DocumentContext")),
+            )
+
+            const getInitialPropsBody = b.blockStatement([
```

Comment:
this is another thing that my refactor branch helps with - we'll be able to pass in a template string with valid TS in it, and the output will be the right AST nodes. might be worth waiting for that change to land